inference/core/devices/utils.py

```diff
@@ -136,5 +137,27 @@ def get_inference_server_id():
         return "UNKNOWN"
 
 
+def get_global_device_id():
+    cache_dir = f"{MODEL_CACHE_DIR}/_config"
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
+    else:
+        with open(cache_dir, "r") as f:
+            config = json.loads(f.read())
+            if config.get("global_device_id") is not None:
+                return config["global_device_id"]
+    global_device_id = DEVICE_ID if DEVICE_ID is not None else get_device_hostname()
+    with open(cache_dir, "w") as f:
```

Comment:
There is a low risk of ending up with garbage in that file due to race condition. We could consider locking https://docs.python.org/3/library/fcntl.html#fcntl.flock

```python
with open(cache_dir, "w") as f:
        fcntl.flock(f, fcntl.LOCK_EX)
        try:
            # (...)
        except Exception as exc:
            # unsuccessful write attempt
        fcntl.flock(f, fcntl.LOCK_UN)
```